util/argo/argo.go

```diff
@@ -763,3 +764,11 @@ func AppInstanceNameFromQualified(name string, defaultNs string) string {
 func ErrProjectNotPermitted(appName, appNamespace, projName string) error {
 	return fmt.Errorf("application '%s' in namespace '%s' is not permitted to use project '%s'", appName, appNamespace, projName)
 }
+
+// TruncateLabelName truncates a label name to a maximum length of characters specified in validation.LabelValueMaxLength
+func TruncateLabelName(name string) string {
+	if len(name) > validation.LabelValueMaxLength {
+		return name[:validation.LabelValueMaxLength]
+	}
+	return name
+}
```

Comment:
If I recall correctly, labels can't end with hyphens. Should we also have logic to strip trailing hyphens after truncating?

Reply:
Alright, let me implement that.

Reply:
Added function to trim the trailing hyphens. Please check.